pkg/server/server_test.go

```diff
@@ -51,47 +55,97 @@ func (b *FakeBusinessLogic) Unbind(request *osb.UnbindRequest, w http.ResponseWr
 }
 
 func (b *FakeBusinessLogic) ValidateBrokerAPIVersion(version string) error {
-	return nil
+	return b.validateAPIVersion(version)
 }
 
 func (b *FakeBusinessLogic) Update(request *osb.UpdateInstanceRequest, w http.ResponseWriter, r *http.Request) (*osb.UpdateInstanceResponse, error) {
 	return b.update(request, w, r)
 }
 
+func defaultValidateFunc(_ string) error {
+	return nil
+}
+
+func strPtr(s string) *string {
+	return &s
+}
+
 func TestGetCatalog(t *testing.T) {
-	expectedResponse := &osb.CatalogResponse{Services: []osb.Service{
+	okResponse := &osb.CatalogResponse{Services: []osb.Service{
 		{
 			Name: "foo",
 		},
 	}}
 
-	api := &rest.APISurface{
-		BusinessLogic: &FakeBusinessLogic{
-			getCatalog: func(w http.ResponseWriter, r *http.Request) (*osb.CatalogResponse, error) {
-				return expectedResponse, nil
+	cases := []struct {
+		name         string
+		validateFunc func(string) error
+		catalogFunc  func(w http.ResponseWriter, r *http.Request) (*osb.CatalogResponse, error)
+		response     *osb.CatalogResponse
+		err          error
+	}{
+		{
+			name: "OK",
+			catalogFunc: func(w http.ResponseWriter, r *http.Request) (*osb.CatalogResponse, error) {
+				return okResponse, nil
+			},
+			response: okResponse,
+		},
+		{
+			name: "version validation error",
+			validateFunc: func(string) error {
+				return errors.New("oops")
+			},
+			err: osb.HTTPStatusCodeError{
+				StatusCode:  http.StatusPreconditionFailed,
+				Description: strPtr("oops"),
 			},
 		},
 	}
 
-	s := New(api)
-
-	fs := httptest.NewServer(s.Router)
-	defer fs.Close()
-
-	config := osb.DefaultClientConfiguration()
-	config.URL = fs.URL
-
-	client, err := osb.NewClient(config)
-	if err != nil {
-		t.Error(err)
-	}
-
-	actualResponse, err := client.GetCatalog()
-	if err != nil {
-		t.Error(err)
-	}
-
-	if !reflect.DeepEqual(actualResponse, expectedResponse) {
-		t.Errorf("Unexpected response\n\nExpected: %#+v\n\nGot: %#+v", expectedResponse, actualResponse)
+	for i := range cases {
+		tc := cases[i]
+		t.Run(tc.name, func(t *testing.T) {
+			validateFunc := defaultValidateFunc
+			if tc.validateFunc != nil {
+				validateFunc = tc.validateFunc
+			}
+
+			api := &rest.APISurface{
+				BusinessLogic: &FakeBusinessLogic{
+					validateAPIVersion: validateFunc,
+					getCatalog:         tc.catalogFunc,
+				},
+			}
+
+			s := New(api)
+			fs := httptest.NewServer(s.Router)
+			defer fs.Close()
+
+			config := osb.DefaultClientConfiguration()
+			config.URL = fs.URL
+
+			client, err := osb.NewClient(config)
+			if err != nil {
+				t.Error(err)
+			}
+
+			actualResponse, err := client.GetCatalog()
+			if err != nil {
+				if tc.err != nil {
+					if e, a := tc.err, err; !reflect.DeepEqual(e, a) {
+						t.Errorf("unexpected error; expected %v, got %v", e, a)
```

Comment:
Nitpick: Should be uppercase:
`Unexpected error...`

Reply:
Fixed